libs/sound/src/PortAudioSound.cpp

```diff
@@ -310,15 +335,39 @@ bool PortAudioSound::Stop()
     return Pa_StopStream(m_stream) == paNoError;
 }
 
+bool PortAudioSound::Close()
+{
+    PaError err;
+    if (0 != (err = Pa_CloseStream(m_stream)))
+    {
+        wxLogError("PortAudio; cannot Close: %s", Pa_GetErrorText(err));
+        return false;
+    }
+    m_stream = NULL;
+    m_isAsynch = false;
+    m_isPaInitialized = false;
+    err = Pa_Initialize();
+    if (err != paNoError) {
+        wxLogError("PortAudio; cannot initialize: %s", Pa_GetErrorText(err));
+        return false;
+    }
+    m_isPaInitialized = true;
+    return true;
+}
+
 
 int PortAudioSound::SoundCallback(void *outputBuffer,
                                   unsigned long framesPerBuffer,
                                   const PaStreamCallbackTimeInfo* timeInfo,
                                   PaStreamCallbackFlags statusFlags)
 {
+    float SoundVolume = (float)((float)g_SoundVolume / (float) 10.0);
```

Comment:
how many casts is  required, really? I would say one...

Reply:
I have had so many troubles with float an cast ... So I am for sure that it works.
change as you like.

Reply:
It doesn't work that way, I cannot change your code. But to cast the dividend is enough, for sure.